firefox_ui_tests/remote/security/test_dv_certificate.py

```diff
@@ -0,0 +1,106 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from marionette_driver import By, Wait
+
+from firefox_ui_harness.decorators import skip_under_xvfb
+from firefox_ui_harness.testcase import FirefoxTestCase
+
+
+class TestDVCertificate(FirefoxTestCase):
+
+    def setUp(self):
+        FirefoxTestCase.setUp(self)
+        self.url = "https://ssl-dv.mozqa.com"
+        self.locationbar = self.browser.navbar.locationbar
+        self.identity_popup = self.browser.navbar.locationbar.identity_popup
+        self.page_info_window = None
+
+    def tearDown(self):
+        self.marionette.set_context("chrome")
+        try:
+            if self.page_info_window is not None:
+                self.page_info_window.close()
+            self.browser.switch_to()
+            self.identity_popup.close(force=True)
+            self.windows.close_all([self.browser])
+        finally:
+            FirefoxTestCase.tearDown(self)
+
+    @skip_under_xvfb
+    def test_dv_cert(self):
+
+        with self.marionette.using_context('content'):
+            self.marionette.navigate(self.url)
+
+        cert = self.browser.tabbar.selected_tab.certificate
+
+        favicon_hidden = self.marionette.execute_script("""
+          return arguments[0].hasAttribute("hidden");
+        """, script_args=[self.locationbar.favicon])
+
+        self.assertFalse(favicon_hidden)
+
+        self.assertIn('identity-icons-https',
+                      self.locationbar.favicon.value_of_css_property('list-style-image'),
+                      'There is a lock icon')
+
+        self.assertEqual(self.identity_popup.box.get_attribute('className'),
+                         'verifiedDomain',
+                         'Identity is verified')
+
+        self.identity_popup.box.click()
+        Wait(self.marionette).until(
+            lambda _: self.identity_popup.is_open,
+            message="The popup should be open"
+        )
+
+        self.assertEqual(self.identity_popup.popup.get_attribute('className'),
+                         'verifiedDomain',
+                         'The Larry UI is domain verified (aka Blue)')
+
+        encryption_icon = self.identity_popup.encryption_icon
+        self.assertNotEqual(encryption_icon.value_of_css_property('list-style-image'),
+                            'none',
+                            'There is a lock icon')
+
+        # Bug 443116
+        # Larry strips the 'www.' from the CName using the eTLDService
+        # This is expected behaviour for the time being
+        expected_name = self.marionette.execute_script("""
+           let Cu = Components.utils;
+           Cu.import("resource://gre/modules/Services.jsm");
+           return Services.eTLD.getBaseDomainFromHost(arguments[0]);
+        """, script_args=[cert['commonName']])
```

Comment:
This method is available now in the security module. So please update this code.